staging/src/kubevirt.io/client-go/apis/core/v1/types.go

```diff
@@ -1890,6 +1890,21 @@ type StopOptions struct {
 	DryRun []string `json:"dryRun,omitempty" protobuf:"bytes,2,rep,name=dryRun"`
 }
 
+// MigrateOptions may be provided on migrate request.
+//
+// +k8s:openapi-gen=true
+type MigrateOptions struct {
+	metav1.TypeMeta `json:",inline"`
+	// When present, indicates that modifications should not be
+	// persisted. An invalid or unrecognized dryRun directive will
+	// result in an error response and no further processing of the
+	// request. Valid values are:
+	// - All: all dry run stages will be processed
+	// +optional
+	// +listType=atomic
+	DryRun []string `json:"dryRun,omitempty" protobuf:"bytes,1,rep,name=dryRun"`
```

Comment:
Can you explain why this needs to be of an array type? why not a regular string?

Reply:
Thanks. Sure, I used the same format used by [kubernetes](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L529), following the same rules. This, imo, prevent the need to wrap the single string into a slice when propagating it toward k8s APIs, and also helps in case of future developments. Let me know if you disagree, we can talk about it :)

Reply:
I wonder why Kubernetes defined it this way..
Anyways you're right :) thanks

Reply:
Just a guess: from the description seems that maybe in the future, you can specify in which stages perform a dry-run. :) Thanks